controller/userController.js

```diff
@@ -0,0 +1,387 @@
+const { validationResult } = require('express-validator');
+const bcrypt = require('bcryptjs');
+const User = require('../models/user');
+const connectDB=require('../connection/database')
+const AccessToken=require('../models/AccessToken')
+const auth=require('../middleware/auth')
+const mongoose =require('mongoose');
+const jwt=require('jsonwebtoken');
+const nodeMailer=require('nodemailer');
+const {EMAIL,PASSWORD}=require('./env');
+const cloudnairy=require('cloudinary').v2;
+const multer=require('multer');
+
+connectDB();
+
+
+const registerUser = async (req, res) => {
+  try {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({ errors: errors.array() });
+    }
+
+    const { username, password, confirmPassword, email, firstname, lastname } = req.body;
+
+    const userExist = await User.findOne({ username });
+
+    if (userExist) {
+      return res.status(409).json({ error: 'Username already exists.' });
+    }
+
+    const emailExist = await User.findOne({ email });
+    if (emailExist) {
+      return res.status(409).json({ error: 'Email already exists.' });
+    }
+
+    if (password !== confirmPassword) {
+      return res.status(400).json({ error: 'Passwords do not match.' });
+    }
+
+    const hashPass = await bcrypt.hash(password, 10);
+
+    const newUser = new User({
+      username: username,
+      password: hashPass,
+      email: email,
+      firstname: firstname,
+      lastname: lastname,
+    });
+
+    await newUser.save();
+    
+    let config={
+           service:'gmail',
+           auth:{
+            user:EMAIL,
+            pass:PASSWORD
+           }
+    }
+    let transporter=nodeMailer.createTransport(config);
+
+    const mailOptions={
```

Comment:
explicitly use the functions  for specific logic so that its more readable and concise